frontend/src/components/search/SearchBar.tsx

```diff
@@ -1,179 +1,265 @@
+          name: new Date(d.name)?.getFullYear(),
+        }))
+        .filter(
+          (obj, index, self) =>
+            index === self.findIndex((o) => o.name === obj.name)
+        ),
+    },
+    sectors: {
+      field: "sector",
+      description: "road, rail, aviation, water transportation",
+      options: data?.facets?.sectors?.items,
+      isMultiple: true,
+    },
+    modes: {
+      field: "mode",
+      description: "car, 2W, 3W, multi-modal etc.",
+      options: data?.facets?.modes?.items,
+      isMultiple: true,
+    },
+    services: {
+      field: "service",
+      description: "passenger or freight",
+      options: data?.facets?.services?.items,
+      isMultiple: true,
+    },
+  };
+
+  /* handle click outside the search container to close command list */
   useEffect(() => {
     const handleClickOutside = (event: MouseEvent) => {
       if (
         commandRef.current &&
         !commandRef.current.contains(event.target as Node)
       ) {
-        if (!hideDatasetSuggestion) setShowCommandList(false);
+        setShowCommandList(false);
       }
     };
-
     document.addEventListener("mousedown", handleClickOutside);
     return () => {
       document.removeEventListener("mousedown", handleClickOutside);
     };
   }, []);
+  //get stored searches
+  useEffect(() => {
+    if (typeof window !== "undefined")
+      setStoredSearches(
+        JSON.parse(localStorage?.getItem("tdcRecentSearches") ?? "[]")
+      );
+  }, []);
 
-  const handleNarrowSelect = (facet: any) => {
-    if (facet) {
-      onChange([{ value: facet, key: facetName! }]);
-      queryInputRef.current?.focus();
-      // setFocus("query");
-    }
+  useEffect(() => {
+    const setQueryValue = setTimeout(() => {
+      setQuery(inputValue);
+    }, 400);
+
+    return () => clearTimeout(setQueryValue);
+  }, [inputValue]);
+
+  const handleFacetNameChange = (facetName: string) => {
+    if (facetName) setFacetName(facetName);
+  };
+
+  const handleFacetValueChange = (item: FacetValueProps) => {
+    setFacetValue(item);
+    inputRef?.current?.focus();
   };
 
   const handleTyping = (value: string) => {
-    setInternalQuery(value);
+    setInputValue(value);
     setIsTyping(value.length > 0);
   };
 
-  useEffect(() => {
-    if (!internalQuery) setInternalQuery(query);
-  }, [query]);
-
   const handleCancelSearch = () => {
-    setInternalQuery("");
-    const changes: any = [];
-    if (facetName) changes.push({ value: undefined, key: facetName });
-    changes.push({ value: undefined, key: "query" });
-    onChange(changes);
-    setTimeout(() => {
-      queryInputRef.current?.focus();
-      // setFocus("query");
-    }, 150);
+    setFacetName("");
+    setFacetValue({
+      display_name: "",
+      name: "",
+    });
+    setInputValue("");
+    setQuery("");
+    setIsTyping(false);
+  };
+
+  const handleSubmit = () => {
+    const value = facets[facetName]?.isMultiple
```

Comment:
You must create the search query based on the search page
![image](https://github.com/user-attachments/assets/b9462699-92f3-4589-9bc6-901aab980aaf)

![image](https://github.com/user-attachments/assets/45ad45d0-8372-4135-85d1-babb9a7be3fb)

Reply:
fixed